src/main/java/com/internet/shop/dao/OrderDao.java

```diff
@@ -0,0 +1,18 @@
+package com.internet.shop.dao;
+
+import com.internet.shop.model.Order;
+import com.internet.shop.model.ShoppingCart;
+import java.util.List;
+import java.util.Optional;
+
+public interface OrderDao {
+    Order completeOrder(ShoppingCart shoppingCart);
```

Comment:
this method should be at the service layer
at the dao we should have create() instead

Reply:
Thx. Fixed.